data/maps/MP_Scenarios/Island_Hopping.wmf/scripting/hop_island.lua

```diff
@@ -14,9 +14,17 @@ function add_workers(hq, workers)
 end
 
 function add_soldiers(hq, soldiers)
-   local setpoints = {}
+   local setpoints = hq:get_soldiers("all")
    for sdescr, count in pairs(soldiers) do
-      setpoints[ {sdescr:match("(%d):(%d):(%d):(%d)")} ] = count
+      local desc = {sdescr:match("(%d):(%d):(%d):(%d)")}
+      local did_add = false
+      for key, value in pairs(setpoints) do
+         if key[1] == desc[1] and key[2] == desc[2] and key[3] == desc[3] and key[4] == desc[4] then
+            setpoints[desc] = value + count
```

Comment:
Also this:

```suggestion
            setpoints[key] = value + count
```

I think the problem here was that because `desc` and `key` are separate tables, they become different indexes.